core-document-store/src/main/java/org/gluu/service/document/store/provider/DBDocumentStoreProvider.java

```diff
@@ -0,0 +1,229 @@
+package org.gluu.service.document.store.provider;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.Charset;
+import java.util.Base64;
+import java.util.Date;
+import java.util.Properties;
+
+import javax.annotation.PostConstruct;
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+import org.apache.commons.io.IOUtils;
+import org.gluu.persist.PersistenceEntryManager;
+import org.gluu.persist.ldap.impl.LdapEntryManager;
+import org.gluu.persist.ldap.impl.LdapEntryManagerFactory;
+import org.gluu.service.document.store.conf.DBDocumentStoreConfiguration;
+import org.gluu.service.document.store.conf.DocumentStoreConfiguration;
+import org.gluu.service.document.store.conf.DocumentStoreType;
+import org.gluu.service.document.store.service.DBDocumentService;
+import org.gluu.service.document.store.service.OxDocument;
+import org.gluu.util.StringHelper;
+import org.gluu.util.security.StringEncrypter;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * @author Shekhar L. on 26/04/2022
+ */
+@ApplicationScoped
+public class DBDocumentStoreProvider extends DocumentStoreProvider<DBDocumentStoreProvider> {
+
+	@Inject
+	private Logger log;
+	
+
+	@Inject
+	private DBDocumentService documentService;
+	
+	@Inject
+	private DocumentStoreConfiguration documentStoreConfiguration;
+
+	private DBDocumentStoreConfiguration dbDocumentStoreConfiguration;
+	
+	
+	@Inject
+	private PersistenceEntryManager persistenceEntryManager;
+
+    public DBDocumentStoreProvider() {
+	}
+
+	public DocumentStoreType getProviderType() {
+		return DocumentStoreType.DB;
+	}
+	
+	@PostConstruct
+	public void init() {
+		this.dbDocumentStoreConfiguration = documentStoreConfiguration.getDbConfiguration();
+	}
+	
+	public void configure(DocumentStoreConfiguration documentStoreConfiguration) {
+		this.log = LoggerFactory.getLogger(DBDocumentStoreProvider.class);
+		this.documentStoreConfiguration = documentStoreConfiguration;
+		this.persistenceEntryManager = dbDocumentStoreConfiguration.getPersistenceEntryManager();
```

Comment:
Instead of attempting to create LDAP connection pool we need to get `persistenceEntryManager` as additional parameter of `configure` method. It's standalone application responsibility to provide it